### PR TITLE
Prepare forms for TOCs

### PR DESCRIPTION
**Pull request by deiferni**
Depends on https://github.com/4teamwork/plonetheme.teamraum/pull/477.

This PR prepares forms that are required to generate a TOC or to manage the period.

- Add a field for a toc template on commitee-container and committee
- Make Periods editable

It also contains an attempt to use the form/formsupport base-classes whenever possible.

*The CL entry for all these changes will follow in a later PR (in order to avoid splitting the CL entries too much)*

Part of #1373.

### CODE REVIEW

File: opengever/meeting/tests/test_toc.py [hunk 1]
```diff
@@ -75,37 +76,48 @@ def setUp(self):
...
-            'dossier_reference_number': u'1.1.4 / 1',
-            'repository_folder_title': u'Business',
-            'meeting_date': 'Jan 01, 2010',
-            'decision_number': 2,
-            'has_proposal': True,
-            'meeting_start_page_number': 33,
+            'group_title': u'5',
+            'contents': [
+                {
+                    'title': u'5 Dinge',
```
Comment (phgross) [inline opengever/meeting/tests/test_toc.py]: Please add an item with a starting umlaut in the title.

File: opengever/meeting/tests/test_committee_container.py [hunk 2]
```diff
@@ -33,6 +34,15 @@ def test_supports_translated_title(self, browser):
         browser.find('DE').click()
         self.assertEquals(u'Sitzungen', browser.css('h1').first.text)
 
+    def test_get_toc_template(self):
+        container = create(
+            Builder('committee_container').having(
+                protocol_template=self.template,
```
Comment (phgross) [inline opengever/meeting/tests/test_committee_container.py]: Test is not realy significant, remove protocol_template and excerpt_tempalte or use a different tempalte for the toc ...

File: opengever/meeting/model/membership.py [hunk 3]
```diff
@@ -64,25 +62,14 @@ def _format_date(self, date):
...
-    def get_url(self, context):
-        return "{}/membership-{}".format(context.absolute_url(),
-                                         self.membership_id)
-
-    def get_edit_url(self, context):
-        return '/'.join((self.get_url(context), 'edit'))
+    def get_url(self, context, view=None):
+        parts = [
+            context.absolute_url(),
+            '/membership-{}'.format(self.membership_id),
```
Comment (phgross) [inline opengever/meeting/model/membership.py]: This method returns a url with two slashes before `membership-` ...

Comment (phgross) [inline opengever/meeting/model/membership.py]: Add a short test

File: opengever/meeting/browser/periods.py [hunk 4]
```diff
@@ -185,3 +190,12 @@ def __init__(self, *args, **kwargs):
 
     def available(self):
         return is_meeting_feature_enabled()
+
+
+class EditPeriod(ModelEditForm):
+
+    fields = field.Fields(IPeriodModel)
+
```
Comment (phgross) [inline opengever/meeting/browser/periods.py]: two empty lines ✂️

File: opengever/meeting/model/period.py [hunk 5]
```diff
@@ -46,10 +46,20 @@ class Period(Base):
...
+        return True
+
+    def is_removable(self):
+        return False
+
     def get_title(self):
         return u'{} ({} - {})'.format(
             self.title, self.get_date_from(), self.get_date_to())
 
+    def get_url(self, context, view=None):
```
Comment (phgross) [inline opengever/meeting/model/period.py]: We create on every SQL Object with a wrapper implementation a property `wrapper_id` i would keep that an use them in the `get_url` method

```
@property
def wrapper_id(self):
    return 'meeting-{}'.format(self.meeting_id)
```

Comment (phgross) [inline opengever/meeting/model/period.py]: Never mind, it's changed in a later comment 👍

**Comment by deiferni** [comment]
@phgross updated

**Pull request merged**